independent-projects/tools/analytics-common/src/test/java/io/quarkus/analytics/rest/RestClientFailTest.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.analytics.rest;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static io.quarkus.analytics.common.ContextTestData.createContext;
+import static io.quarkus.analytics.rest.RestClient.IDENTITY_ENDPOINT;
+import static io.quarkus.analytics.rest.RestClient.TTL;
+import static io.quarkus.analytics.util.StringUtils.getObjectMapper;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.tomakehurst.wiremock.WireMockServer;
+
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.Identity;
+
+class RestClientFailTest {
+
+    private static final int MOCK_SERVER_PORT = 9300;
+    private static final String TEST_CONFIG_URL = "http://localhost:" + MOCK_SERVER_PORT + "/" + "config";
+    private static final WireMockServer wireMockServer = new WireMockServer(MOCK_SERVER_PORT);
+
+    @BeforeAll
+    static void start() throws JsonProcessingException {
+        wireMockServer.start();
+        wireMockServer.stubFor(post(urlEqualTo("/" + IDENTITY_ENDPOINT))
+                .willReturn(aResponse()
+                        .withStatus(201)
+                        .withFixedDelay(TTL + 50) // will take too long to respond
```

Comment:
Can you make the TTL configurable so this won't delay the test for 2050ms (same below)?

Reply:
I was testing that elsewhere, but I can make it configurable.

Reply:
added quarkus.analytics.timeout